jib-gradle-plugin/src/integration-test/java/com/google/cloud/tools/jib/gradle/SingleProjectIntegrationTest.java

```diff
@@ -195,10 +209,22 @@ public void testBuild_simple() throws IOException, InterruptedException, DigestE
               "No classes files were found - did you compile your project?"));
     }
 
+    String output = JibRunHelper.buildAndRun(simpleTestProject, targetImage);
+
     Assert.assertEquals(
         "Hello, world. An argument.\n1970-01-01T00:00:01Z\nrw-r--r--\nrw-r--r--\nfoo\ncat\n"
             + "1970-01-01T00:00:01Z\n1970-01-01T00:00:01Z\n",
-        JibRunHelper.buildAndRun(simpleTestProject, targetImage));
+        output);
+
+    String digest =
+        readDigestFile(simpleTestProject.getProjectRoot().resolve("build/jib-image.digest"));
+    String imageReferenceWithDigest = ImageReference.parse(targetImage).withTag(digest).toString();
+    Assert.assertEquals(output, JibRunHelper.pullAndRunBuiltImage(imageReferenceWithDigest));
```

Comment:
Now that you run `pullAndRunBuiltImage()`, I think you don't need `new Command("docker", "run", ...)` below anymore. And the order would be less confusing if we do

```java
    String digest = readDigestFile(simpleTestProject.getProjectRoot().resolve("build/jib-image.digest"));
    String id = readDigestFile(simpleTestProject.getProjectRoot().resolve("build/jib-image.id"));
    Assert.assertNotEquals(digest, id);

    String imageReferenceWithDigest = ImageReference.parse(targetImage).withTag(digest).toString();
    Assert.assertEquals(output, JibRunHelper.pullAndRunBuiltImage(imageReferenceWithDigest))
```
```

Reply:
The second `new Command("docker", "run", ...)` is to test running with the generated image id; this one tests the digest.

Reply:
Ah, now I get it. Cool!